rundk/main.go

```diff
@@ -0,0 +1,192 @@
+	pflag.BoolVar(&useLocalKubeconfig, "use-local-kubeconfig", false, "Use the same kubeconfig as local, which can be set "+
+		"either by setting env var KUBECONFIG or from ~/.kube/config")
+	pflag.StringSliceVar(&mounts, "mounts", []string{}, "A list of extra folders or files separated by comma that need to be mounted to run the test flow."+
+		"It must be in the format of `source1:target1,source2:target2,source3:target3`.")
+	pflag.StringSliceVar(&mandatoryEnvVars, "mandatory-env-vars", []string{}, "A list of env vars separated by comma that must be set on local.")
+	pflag.StringSliceVar(&optionalEnvVars, "optional-env-vars", []string{}, "A list of env vars separated by comma that optionally need to be set on local.")
+	pflag.Parse()
+
+	cmd, cancel := setup()
+	defer cancel()
+
+	run(cmd, pflag.Args()...)
+}
+
+func setup() (interactive.Docker, func()) {
+	var err error
+
+	// Until everyone is using 20.xx version of docker (see https://github.com/docker/cli/pull/1498) adding the --pull flag,
+	//  need to separately pull the image first to be sure we have the latest
+	pull := interactive.NewCommand("docker", "pull", image)
+	if err = pull.Run(); err != nil {
+		log.Fatal("Error pulling the test image: ", err)
+	}
+
+	builtUpDefers := make([]func(), 0)
+
+	// Setup command
+	cmd := interactive.NewDocker()
+	envs := interactive.Env{}
+
+	// Setup temporary directory to save the artifacts
+	tmpDir, err := ioutil.TempDir("/tmp", "prow-docker.")
+	if err != nil {
+		log.Fatal("Error setting up the temporary directory: ", err)
+	}
+
+	// Setup docker-in-docker
+	if dindEnabled {
+		cmd.AddMount("bind", "/var/run/docker.sock", "/var/run/docker.sock")
+		// On a MAC machine, set to host networking to make DinD work. Not
+		// required on Linux.
+		if runtime.GOOS == "darwin" {
+			cmd.AddArgs("--network", "host")
+		}
+	}
+
+	// Setup gcloud credentials
+	if useLocalGcloudCredentials {
+		gcloudKey := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
+		// If GOOGLE_APPLICATION_CREDENTIALS is not empty, also mount the key file
+		// to the container
+		if gcloudKey != "" {
+			envs["GOOGLE_APPLICATION_CREDENTIALS"] = gcloudKey
+			if _, err := os.Stat(gcloudKey); !os.IsNotExist(err) {
+				cleanup := cmd.CopyAndAddMount("bind", tmpDir, gcloudKey, gcloudKey)
+				builtUpDefers = append(builtUpDefers, cleanup)
+			}
+		}
+		// Copy and mount for gcloud default credentials to be available
+		defaultGcloudConfigPath := path.Join(os.Getenv("HOME"), ".config/gcloud")
+		if _, err := os.Stat(defaultGcloudConfigPath); !os.IsNotExist(err) {
+			cleanup := cmd.CopyAndAddMount("bind", tmpDir, defaultGcloudConfigPath, "/root/.config/gcloud")
+			builtUpDefers = append(builtUpDefers, cleanup)
+		}
+	}
+
+	// Setup kubeconfig
+	if useLocalKubeconfig {
+		kubeconfig := os.Getenv("KUBECONFIG")
+		// If KUBECONFIG is not empty, also mount the kubeconfig files to the
+		// container
+		if kubeconfig != "" {
+			envs["KUBECONFIG"] = kubeconfig
```

Comment:
ditto filename

Reply:
Similar as the above comment related to `GOOGLE_APPLICATION_CREDENTIALS`. Also the `KUBECONFIG` can be pointed to multiple kubeconfig files (for multiple clusters), so it's not quite easy to use constant filenames here.